server/src/main/java/org/opentosca/toscana/plugins/kubernetes/docker/dockerfile/builder/DockerfileBuilder.java

```diff
@@ -0,0 +1,200 @@
+
+     @param command the command to add to the dockerfile
+     @return the object itself to allow chaining (not needed though).
+     */
+    public DockerfileBuilder run(String command) {
+        entries.add(new RunCommand(command));
+        return this;
+    }
+
+    /**
+     Adds a <code>ENTRYPOINT</code> command to the Dockerfile
+
+     @param args If only one command is given <code>ENTRYPOINT <args[0]></code> will
+     be added to the dockerfile otherwise
+     <code>ENTRYPOINT ["<args[0]>","<args[1]>"...]</code> will be added otherwise
+     @return the object itself to allow chaining (not needed though).
+     */
+    public DockerfileBuilder entrypoint(String... args) {
+        entries.add(new EntrypointCommand(args));
+        return this;
+    }
+
+    /**
+     Adds a <code>EXPOSE</code> command for the given port to the dockerfile
+
+     @param port The port to expose
+     @return the object itself to allow chaining (not needed though).
+     */
+    public DockerfileBuilder expose(int port) {
+        entries.add(new ExposeCommand(port));
+        return this;
+    }
+
+    /**
+     Adds a <code>VOLUME</code> command for the given path to the dockerfile
+
+     @param path the path at wich the volume should be created within the Container Image
+     @return the object itself to allow chaining (not needed though).
+     */
+    public DockerfileBuilder volume(String path) {
+        entries.add(new VolumeCommand(path));
+        return this;
+    }
+
+    /**
+     Adds a <code>WORKDIR</code> command for the given path to the dockerfile
+
+     @param directory the path at wich the working directory should be set within the Container Image
+     @return the object itself to allow chaining (not needed though).
+     */
+    public DockerfileBuilder workdir(String directory) {
+        entries.add(new WorkdirCommand(directory));
+        return this;
+    }
+
+    /**
+     Adds a <code>ENV</code> command to the dockerfile
+
+     @param variableName The name of the environment variable
+     @param value        the corresponding value
+     @return the object itself to allow chaining (not needed though).
+     */
+    public DockerfileBuilder env(String variableName, String value) {
+        entries.add(new EnvCommand(variableName, value));
+        return this;
+    }
+
+    /**
+     Enables the compression of run commands to one run command. this means that they get linked using <code>&&</code>
+     <p>
+     <b>Warning:</b> Using this command does not ensure the same behaviour as two sperated run commands.
+     (the current directory ist not reset to the current WORKDIR)
+
+     @return the object itself to allow chaining (not needed though).
+     */
+    public DockerfileBuilder compressRunCommands() {
+        compress = true;
+        return this;
+    }
+
+    /**
+     Writes the dockerfile to the path defined in the constructor
+     */
+    public void build() throws IOException {
```

Comment:
I would rename this to `write()` or `create()` or similar.
Reason: This is not the builder pattern. You don't return a Dockerfile. This is confusing for the developer (especially since the whole EffectiveModel uses the builder pattern)